### ISSUE TITLE
[BUG] `pyproject` linter is not compatible with dependabot

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#5450]
It seems the recent `pyproject` linter is not compatible with dependabot:

https://github.com/sktime/sktime/pull/5443

FYI @yarnabrina

**Comment by yarnabrina** [comment]
I wasn't familiar with this. Asked for better configuration options.

1. https://stackoverflow.com/q/77318146/11117265
2. https://github.com/tox-dev/pyproject-fmt/issues/135

Do you want to handle this quickly? One possibility can be to auto-commit the change suggested by hook first and retrigger CI. I never done this,  but I believe @achieveordie has done similar in the contributors update workflow.

**Comment by fkiraly** [comment]
> Do you want to handle this quickly?

I wouldn't have a solution at the top of my head which is better than reverting the linter...

**Comment by yarnabrina** [comment]
I do think the linter is a good way to ensure consistent formats.

Created #5453 removing the newly added hooks, but I hope this is temporary. My preference will be to add these back once someone suggests a compatible configuration for either dependabot or pyproject-fmt.

**Comment by yarnabrina** [comment]
According to @gaborbernat's reply in https://github.com/tox-dev/pyproject-fmt/issues/135#issuecomment-1773132304, `pyproject-fmt` is following `packaging` in the linting style. Based on my understanding, this is the argument:

```pycon
>>> 
>>> from packaging.specifiers import SpecifierSet
>>> str(SpecifierSet(">=2,<2.1"))
'<2.1,>=2'
>>> 
```

This means we can use at most one of `pyproject-fmt` or `dependabot`. There is `renovate` (may be others, not sure), but I don't know what does it follow.

@fkiraly how do you suggest we address it? If it is to simply not add `pyproject-fmt` in `pre-commit` and keep it that way, we can close this issue to keep status quo.

**Comment by fkiraly** [comment]
I think the benefit of dependabot outweights the benefit of pyproject linting, if we are really forced to choose between the two...

We could trigger the pyproject lint manually - perhaps together with contributors.md update - in a CRON, would that be a way to sidestep the issue?

**Comment by yarnabrina** [comment]
Yes, that can be done. But as I said, I'm not sure how to commit as part of a workflow (by what user credentials) and then re-trigger another workflow.

I think @achieveordie did this for contributors, can you please share some suggestions?
